src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -209,3 +210,32 @@ createEuropeanPhysicalClaim dateToTime isLong strike underlying ccy
     claim = european expiryDate exerciseClaim
   in
     prepareAndTagClaims dateToTime [claim] "Option payment"
+
+-- | Create the claim for a physically settled Dividend option.
+createDividendOptionClaim : (Date -> Time) -> Bool -> Date -> InstrumentQuantity ->
+  Optional InstrumentQuantity -> Optional InstrumentQuantity -> TaggedClaim
+createDividendOptionClaim dateToTime isLong expiryDate cashQuantity
+  sharesQuantity fxQuantity =
+  let
+    cashClaimAmount = scale (Const cashQuantity.amount) (one cashQuantity.unit)
+    cashClaim = if isLong then cashClaimAmount else give cashClaimAmount
+    taggedCashClaims = [("CASH", cashClaim)]
+
+    taggedSharesClaims = case sharesQuantity of
+      None -> []
+      Some shares -> [("SHARES", sharesClaim)]
+        where
+          sharesClaimAmount = scale (Const shares.amount) (one shares.unit)
+          sharesClaim = if isLong then sharesClaimAmount else give sharesClaimAmount
```

Comment:
Trying that now (see separate commit), but I get:
```

    • Couldn't match type ‘(Text,
                            Claim
                              t0
                              (Numeric 10)
                              Daml.Finance.Interface.Types.Common.Types.InstrumentKey
                              o0)’
                     with ‘Claim t1 x0 a0 o1’

```
Could it be that `give` is not defined for tagged claim?

Reply:
I tried something similar originally, but also had problems then, so I went for the separate approach. It would clearly look nicer though if we could apply `give` only once.

Reply:
The following works for me, what do you think?

```suggestion
    taggedCashClaim = ("CASH", cashClaim)

    taggedSharesClaims = case sharesQuantity of
      None -> []
      Some shares -> [("SHARES", sharesClaim)]
        where
          sharesClaim = scale (Const shares.amount) $ one shares.unit

    taggedFxClaims = case fxQuantity of
      None -> []
      Some fx -> [("CASHFX", cashFxClaim)]
        where
          cashFxClaim = scale (Const fx.amount) $ one fx.unit

    claim =
      when (at expiryDate) . orList . fmap (fmap flip) $
      taggedCashClaim :: taggedSharesClaims ++ taggedFxClaims
        where
          flip c = if isLong then c else give c
  in
    prepareAndTagClaims dateToTime [claim] "Option payment"
```